google/resource_compute_target_tcp_proxy.go

```diff
@@ -0,0 +1,171 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/compute/v1"
+)
+
+func resourceComputeTargetTcpProxy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeTargetTcpProxyCreate,
+		Read:   resourceComputeTargetTcpProxyRead,
+		Delete: resourceComputeTargetTcpProxyDelete,
+		Update: resourceComputeTargetTcpProxyUpdate,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"backend_service": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"proxy_header": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "NONE",
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"id": &schema.Schema{
```

Comment:
Could you rename it to `proxy_id`? We are moving away from "id" field. See #399 for more details.